src/components/Chat/ChatSession.js

```diff
@@ -515,6 +564,40 @@ class ChatSession {
     }
   }
 
+  async _describeAndProcessView(item) {
+    const viewDetails = JSON.parse(item.content.data);
+    if (viewDetails.templateType !== InteractiveMessageType.VIEW_RESOURCE) {
+      return;
+    }
+    let newParsedView = {};
+    const ViewResourceInputData = modelUtils.createViewMessageData(viewDetails.data);
+    try {
+      const describeViewResponse = await this.describeView({ viewToken: ViewResourceInputData.viewToken });
+      const newView = describeViewResponse ? describeViewResponse.data.View : {};
+      const Template = JSON.parse(newView.Content.Template);
+      const InputSchema = JSON.parse(newView.Content.InputSchema);
+      newParsedView = {
+        ...newView,
+        Content: {
+          Actions: newView.Content.Actions,
+          Template,
+          InputSchema,
+        },
+        InputData: ViewResourceInputData.viewInputData
+      };
+    } catch (err) {
+      newParsedView = {
+        Content: { InputSchema: {}, Template: {} },
+        ErrorType: 'INVALID_VIEW_ID',
```

Comment:
We're assuming that JSON.parse isn't throwing here

Reply:
The ErrorType used here is to trigger an error view. The logger will display the error message in console.

Reply:
[nit] Can be moved to ChatJS.